simulator/src/main/java/pt/ulisboa/tecnico/socialsoftware/ms/coordination/workflow/ExecutionPlan.java

```diff
@@ -47,30 +61,91 @@ public boolean canExecute(HashMap<FlowStep, CompletableFuture<Void>> stepFutures
     }
 
     public CompletableFuture<Void> execute(UnitOfWork unitOfWork) {
-
+        String stepName;
+        Map<String, List<Integer>> behaviour = BehaviourHandler.getInstance().loadStepsFile(functionalityName);
+        BehaviourHandler.getInstance().appendToReport(reportSteps(behaviour));
+        List<Integer> behaviourValues = new ArrayList<>();
+        
         // Initialize futures for steps with no dependencies
         for (FlowStep step: plan) {
-            if (dependencies.get(step).isEmpty()) {
-                this.stepFutures.put(step, step.execute(unitOfWork)); // execute and save the steps with no dependencies
-                executedSteps.put(step, true);
+            stepName = step.getName();
+
+            // Check if the step is in the behaviour map
+            behaviourValues = behaviour.containsKey(stepName) ? behaviour.get(stepName) : Arrays.asList(1,0,0);
+            System.out.println("Step: " + stepName + " Behaviour: " + behaviourValues);
+            if (behaviourValues.get(0) == 1) {   
+                if (dependencies.get(step).isEmpty()) {
+                    this.stepFutures.put(step, step.execute(unitOfWork)); // Execute and save the steps with no dependencies
+                    executedSteps.put(step, true);
+                }
             }
         }
 
         // Execute steps based on dependencies
         for (FlowStep step: plan) {
-            if (!this.stepFutures.containsKey(step)) { // if the step has dependencies
-                ArrayList<FlowStep> deps = dependencies.get(step); // get all dependencies
-                CompletableFuture<Void> combinedFuture = CompletableFuture.allOf( // create a future that only executes when all the dependencies are completed
-                    deps.stream().map(this.stepFutures::get).toArray(CompletableFuture[]::new) // maps each dependency to its corresponding future in stepFutures
-                );
-                this.stepFutures.put(step, combinedFuture.thenCompose(ignored -> step.execute(unitOfWork))); // only executes after all dependencies are completed
+            stepName = step.getName();
+            behaviourValues = behaviour.containsKey(stepName) ? behaviour.get(stepName) : Arrays.asList(1,0,0);
+            if (behaviourValues.get(0) == 1) { 
```

Comment:
It look like that it should allways be ==1... it is weird, what is the meaning when it is not?